core/src/main/java/org/elasticsearch/index/query/OrQueryBuilder.java

```diff
@@ -107,13 +107,19 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
         }
 
         BooleanQuery query = new BooleanQuery();
+        int addedFilters = 0;
         for (QueryBuilder f : filters) {
             Query innerQuery = f.toQuery(parseContext);
             // ignore queries that are null
             if (innerQuery != null) {
                 query.add(innerQuery, Occur.SHOULD);
+                addedFilters++;
             }
         }
+        if (addedFilters == 0) {
+            // no inner lucene query exists, ignore upstream
+            return null;
+        }
```

Comment:
here too, not sure why we have to return null now, also we can use query.clauses() to count the added filters